packages/components/src/typography/typography.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { ReactNode } from "react";
+import tw, { TwStyle, styled } from "twin.macro";
+
+type TypographyElement = "h1" | "h2" | "h3" | "h4" | "h5" | "h6" | "span" | "p";
+
+type TypographySize =
+  | "h1"
+  | "h2"
+  | "h3"
+  | "h4"
+  | "h5"
+  | "body"
+  | "sm"
+  | "xs"
+  | "button"
+  | "caption"
+  | "overline";
+
+type TypographyColor =
+  | "alert"
+  | "base"
+  | "brand"
+  | "info"
+  | "success"
+  | "warning"
+  | "white";
+
+function elementFromSize(size: TypographySize): TypographyElement {
+  switch (size) {
+    case "h1":
+    case "h2":
+    case "h3":
+    case "h4":
+    case "h5":
+      return size;
+    default:
+      return "p";
+  }
+}
+
+function styleFromColor(color: TypographyColor): TwStyle {
+  switch (color) {
+    case "alert":
+      return tw`text-alert-700`;
+    case "brand":
+      return tw`text-brand-700`;
+    case "info":
+      return tw`text-info-700`;
+    case "success":
+      return tw`text-success-700`;
+    case "warning":
+      return tw`text-warning-700`;
+    case "white":
+      return tw`text-white`;
+    default:
+      return tw`text-font-base`;
+  }
+}
+
+function styleFromSize(size: TypographySize): TwStyle {
+  switch (size) {
+    case "h1":
+      return tw`text-h1 leading-h1 capitalize`;
+    case "h2":
+      return tw`text-h2 leading-h2 capitalize`;
+    case "h3":
+      return tw`text-h3 leading-h3 capitalize`;
+    case "h4":
+      return tw`text-h4 leading-h4 capitalize`;
+    case "h5":
+      return tw`text-h5 leading-body capitalize`;
+    case "body":
+      return tw`text-body leading-body`;
+    case "sm":
+      return tw`text-sm leading-sm`;
+    case "xs":
+      return tw`text-xs leading-xs`;
+    case "button":
+      return tw`text-sm leading-body`;
+    case "caption":
+      return tw`text-xs leading-sm`;
+    case "overline":
+      return tw`text-xs leading-sm uppercase`;
+    default:
```

Comment:
probably don't need default because typescript, feel free to remove this rule from eslint